analyses/cell-type-nonETP-ALL-03/README.md

```diff
@@ -0,0 +1,77 @@
+# Non-ETP T-ALL Annotation (SCPCP000003)
+
+This analysis module will include code to annotate cell types in non-ETP T-ALL from SCPCP000003 (n=11) present on the ScPCA portal.
+
+## Description
+
+We first aim to annotate the cell types in non-ETP T-ALL, and use the annotated B cells in the sample as the "normal" cells to identify tumor cells, since T-ALL is caused by the clonal proliferation of immature T-cell [<https://www.nature.com/articles/s41375-018-0127-8>].
+
+-   We use the cell type marker (`Azimuth_BM_level1.xlsx`) from [Azimuth Human Bone Marrow reference](https://azimuth.hubmapconsortium.org/references/#Human%20-%20Bone%20Marrow). In total, there are 14 cell types: B, CD4T, CD8T, Other T, DC, Monocytes, Macrophages, NK, Early Erythrocytes, Late Erythrocytes, Plasma, Platelet, Stromal, and Hematopoietic Stem and Progenitor Cells (HSPC). Based on the exploratory analysis, we believe that most of the cells in these samples do not express adequate markers to be distinguished at finer cell type level (eg. naive vs memory, CD14 vs CD16 etc.), and majority of the cells should belong to T-cells. In addition, we include the marker genes for cancer cell in immune system from [ScType](https://sctype.app/database.php) database.
```

Comment:
It was created from the website table (combination of `celltype.l2` and `celltype.l1`), and I converted the markers from gene symbol to ensembl ID. Do you mean to provide the URL (https://azimuth.hubmapconsortium.org/references/#Human%20-%20Bone%20Marrow) as part of the header of the file?

I am saving it as an excel file because it is the input format required by ScType, unless I go and change their codes. If I understand the Cell Ontology IDs correctly, it provides an ID for a specific cell type (eg. CL:0000904 for central memory CD4+ T cell). This is the reason I am using `celltype.l1`, because I need the cell type annotation to be very general (eg. just CD4+ T, instead of specifying naive or memory). 
I did try with finer cell type label, but I found out that the cells (e.g. different types of CD4 and CD8 and even progenitors) all cluster together (fail to be separated), thus they cannot be annotated at cluster level. I also try to do annotation at cell level using anchor transfer etc. The prediction results is really bad. Thus, I suspect that they don't express enough markers to be separated at the finer level.

Reply:
Thank you for the clarification. I don't think you need to provide the URL in the header as you do already have it in the readme, but I am surprised that ScType can not take a `.csv` file! (I just looked at the code and I do have an idea for how to get around this: I think you should be able to run something like `gene_sets_prepare(openxlsx::buildWorkbook(marker_df), "Immune system")` to work from any data frame, but this is not something we really need to address now.

As for Cell Ontology IDs, they are hierarchical, so you should be able to use a higher-level term if needed. But I think that you can use the values given as links in the Azimuth table to start: For example for B cells they link to [CL:0000945](http://purl.obolibrary.org/obo/CL_0000945), and for the CD4 T cells it links to [CL:0000624](http://purl.obolibrary.org/obo/CL_0000624), which is a more ID; The central memory CD4 cells, [CL:0000904](http://purl.obolibrary.org/obo/CL_0000904) can be found as a child term to the more general term.

This is the primary benefit of using the ontology, in fact! We can use these terms to harmonize across annotations even if different samples are annotated to different depths by various groups.

Reply:
I made the changes as suggested for using `.csv` file (the codes will be shown up in `scripts/02-03_annotation.R` in next pull request).
As of now, I put the Cell Ontology IDs under the column of `shortName` in `Azimuth_BM_level1.csv`. I don't think there is an ID for "Other T", since it is such a broad category, and I could not find IDs for Early and Late Eryth. I just leave them as how it is now.

Reply:
Actually, I have another concern regarding SAM and ScType. Each time I run SAM, the clustering output changes, although the seed is set as 0 (by default) in `sam$run()`. The log is shown as image below.
![Screenshot 2024-09-24 at 5 12 45 PM](https://github.com/user-attachments/assets/4ad363ef-e084-4e29-8a65-f4b370fd0a1f)

Since ScType annotates by cluster, so the cell annotation may change if they are found in another cluster, with different annotation. This is actually quite likely to happen, especially for those cells that don't have clear markers expressed. Eg. Depending on the clustering results, they may be annotated as `Unknown`, if the score of a cluster is less than a quarter of #cells in that cluster.